apps/lockfile-explorer-web/src/parsing/lfxGraphLoader.ts

```diff
@@ -0,0 +1,363 @@
+  } else if (result.version.startsWith('/')) {
+    result.entryId = result.version;
+  } else if (result.dependencyType === DependencyKind.PEER_DEPENDENCY) {
+    if (node?.peerDependencies) {
+      result.peerDependencyMeta = {
+        name: result.name,
+        version: node.peerDependencies[result.name],
+        optional:
+          node.peerDependenciesMeta && node.peerDependenciesMeta[result.name]
+            ? node.peerDependenciesMeta[result.name].optional
+            : false
+      };
+      result.entryId = 'Peer: ' + result.name;
+    } else {
+      // eslint-disable-next-line no-console
+      console.error('Peer dependencies info missing!', node);
+    }
+  } else {
+    result.entryId = '/' + result.name + '/' + result.version;
+  }
+  return result;
+}
+
+// node is the yaml entry that we are trying to parse
+function parseDependencies(
+  dependencies: LockfileDependency[],
+  lockfileEntry: LockfileEntry,
+  node: ILockfileNode
+): void {
+  if (node.dependencies) {
+    for (const [pkgName, pkgVersion] of Object.entries(node.dependencies)) {
+      dependencies.push(
+        createLockfileDependency(pkgName, pkgVersion, DependencyKind.DEPENDENCY, lockfileEntry)
+      );
+    }
+  }
+  if (node.devDependencies) {
+    for (const [pkgName, pkgVersion] of Object.entries(node.devDependencies)) {
+      dependencies.push(
+        createLockfileDependency(pkgName, pkgVersion, DependencyKind.DEV_DEPENDENCY, lockfileEntry)
+      );
+    }
+  }
+  if (node.peerDependencies) {
+    for (const [pkgName, pkgVersion] of Object.entries(node.peerDependencies)) {
+      dependencies.push(
+        createLockfileDependency(pkgName, pkgVersion, DependencyKind.PEER_DEPENDENCY, lockfileEntry, node)
+      );
+    }
+  }
+  if (node.transitivePeerDependencies) {
+    for (const dep of node.transitivePeerDependencies) {
+      lockfileEntry.transitivePeerDependencies.add(dep);
+    }
+  }
+}
+
+function createLockfileEntry(options: {
+  rawEntryId: string;
+  kind: LockfileEntryFilter;
+  rawYamlData: ILockfileNode;
+  duplicates?: Set<string>;
+  subspaceName?: string;
+}): LockfileEntry {
+  const { rawEntryId, kind, rawYamlData, duplicates, subspaceName } = options;
+
+  const result = new LockfileEntry(kind);
+
+  result.rawEntryId = rawEntryId;
+
+  if (rawEntryId === '.') {
+    // Project Root
+    return result;
+  }
+
+  if (kind === LockfileEntryFilter.Project) {
+    const rootPackageJsonFolderPath = new Path(`common/temp/${subspaceName}/package.json`).dirname() || '';
+    const packageJsonFolderPath = new Path('.').relative(
+      new Path(rootPackageJsonFolderPath).concat(rawEntryId)
+    );
+    const packageName = new Path(rawEntryId).basename();
+
+    if (!packageJsonFolderPath || !packageName) {
+      // eslint-disable-next-line no-console
+      console.error('Could not construct path for entry: ', rawEntryId);
+      return result;
+    }
+
+    result.packageJsonFolderPath = packageJsonFolderPath.toString();
```

Comment:
Sorta feels like all of these properties should be computed before the `result` object is constructed so they all can be made `readonly`.